pkg/controller/attach_detach.go

```diff
@@ -124,11 +125,11 @@ func (r *ReconcileAttachDetach) Reconcile(ctx context.Context, request reconcile
 			return reconcileReturnOnError(ctx, azVolumeAttachment, "detach", err, r.retryInfo)
 		}
 		// attachment request
-	} else if azVolumeAttachment.Status.Detail == nil {
+	} else if azVolumeAttachment.Status.Detail == nil || (azVolumeAttachment.Status.State == azdiskv1beta2.AttachmentPending && azureutils.MapContains(azVolumeAttachment.Status.Annotations, consts.VolumeAttachRetryAnnotation)) {
```

Comment:
If we remove the annotation and reset the state in replica.go, we won't be needing to add any new conditions here.